sw/device/tests/entropy_src_kat_test.c

```diff
@@ -61,29 +106,55 @@ static void entropy_with_fw_override_enable(dif_entropy_src_t *entropy_src) {
  * See:
  * https://csrc.nist.gov/projects/cryptographic-algorithm-validation-program/secure-hashing
  */
-void test_sha384_kat(dif_entropy_src_t *entropy) {
-  CHECK_DIF_OK(dif_entropy_src_set_enabled(entropy, kDifToggleDisabled));
-  entropy_with_fw_override_enable(entropy);
-  CHECK_DIF_OK(dif_entropy_src_conditioner_start(entropy));
+
+void test_sha384_kat(dif_entropy_src_t *entropy_src) {
+  CHECK_DIF_OK(dif_entropy_src_set_enabled(entropy_src, kDifToggleDisabled));
+  entropy_with_fw_override_enable(entropy_src);
+
+  // Though most of the entropy_src state is cleared on disable, the
+  // SHA3 conditioner accumulates entropy even from aborted seeds. For
+  // the KAT though, we must clear any previously absorbed entropy.
+  flush_sha3_conditioner(entropy_src);
 
   const uint32_t kInputMsg[kEntropyFifoBufferSize] = {
       0xa52a0da9, 0xcae141b2, 0x6d5bab9d, 0x2c3e5cc0, 0x225afc93, 0x5d31a610,
       0x91b7f960, 0x0d566bb3, 0xef35e170, 0x94ba7d8e, 0x534eb741, 0x6b60b0da,
   };
-  CHECK_DIF_OK(dif_entropy_src_observe_fifo_write(entropy, kInputMsg,
-                                                  ARRAYSIZE(kInputMsg)));
 
-  CHECK_DIF_OK(dif_entropy_src_conditioner_stop(entropy));
+  CHECK_DIF_OK(dif_entropy_src_conditioner_start(entropy_src));
+
+  dif_result_t op_result;
+  uint32_t fail_count = 0;
+  uint32_t count;
+  uint32_t total = 0;
+
+  // Load the input data.
```

Comment:
this is a bit after the fact, but i honestly think it would have been okay in the test to check for fifo full before doing a write.
That way you wouldn't need to absorb the function into DIF. And I'm guessing since DIF's aren't good for polling (because of the unittest), that's why you added the function to say how many entries were written such that the upper layer can retry. 

@moidx 
@engdoreis thoughts?

Reply:
Not suggesting this be changed now that it's already done, just wondering for the future when similar situations come up. Because we have a lot of blocks whose FIFO access doesn't block on full or empty.

Reply:
Checking for FIFO full before doing the write in this case will only help if you limit the writes to single words.   The FIFO in this case is mighty flimsy. It's actually just a prim-packer FIFO, converting 32b to 64b), and transient single cycle full events happen in verification happen all the time.   

Fun detail: We actually just roll with these events in verification, the scoreboard just notes that FW is not following the API, and note that the user input is going to get an alert even though SHA input data is going to be dropped.  The output SHA digest may not match what the Host Sequence would expect, but since the scoreboard is closely modelling the flow of data through the DUT, including dropped data, we can predict the output data anyway.

Reply:
Also: @tjaychen, apologies if I jumped the gun too fast, I have perhaps a bit overly eager to merge #14644 which has been pending for a while.  I'll be happy to make any changes you request.

Reply:
o no worries. I think this is fine honestly. I just wondered if you had to do extra work because unittests are kind of unfriendly towards polling.  Was that why you did it this way..? or am i just speculating? also wanted to use this chance to get a sense on how to handle future ones.

Reply:
Ha, they are unfriendly toward polling, it is true.  I was trying to just alter the existing interface as little as possible.  The alternative would be to restrict the input size to one word.  This would be a more drastic interface change.

Reply:
Different FIFOs may have better registers for announcing how much space they have also.  This one is particularly annoying in that it can fill even after one write, and there is no feedback otherwise.